mini-projects/00-jq-rs/src/main.rs

```diff
@@ -1,3 +1,61 @@
+use clap::Parser;
+use processor::*;
+use serde_json::Value;
+use std::fs;
+use std::io;
+use std::env;
+use thiserror::Error;
+mod processor;
+#[derive(Parser)]
+struct Args {
+    #[arg(short, long, default_value_t = true)]
+    coloroutput: bool,
+    #[arg(short, long, default_value_t = false)]
+    monochromeoutput: bool,
+    #[arg(short, long, default_value_t = false)]
+    sortkeys: bool,
+    #[arg(short, long, default_value_t = 2)]
+    indent: u32,
+    #[arg(long, default_value_t = false)]
+    compactoutput: bool,
+    needle: String,
+    path: String,
+}
+#[derive(Error, Debug)]
+pub enum ParseError {
+    #[error("File error: {0}")]
+    FileError(#[from] io::Error), // Automatically convert from std::io::Error
+
+    #[error("JSON parsing error: {0}")]
+    JsonError(#[from] serde_json::Error), // Automatically convert from serde_json::Error
+}
 fn main() {
-    println!("Hello World!");
+    let args = Args::parse();
+    let jq_colors: std::result::Result<String, env::VarError> = env::var("JQ_COLORS");
+    //check if we found an env variable
+    let jq_env = match jq_colors {
+        Ok(env) => env,
+        //default value is used if one is not found
+        Err(_) => String::from("0;90:0;37:0;37:0;37:0;32:1;37:1;37:1;34"),
```

Comment:
nit: this string is a little gross, potentially would have been nice to assign each individual parameter to its own constant and then just format it into this string - that way, a dev can quickly scan the variables names to know which one is relevant to them, rather than needing to pull up the documentation